tests/cypress/support/authentication.js

```diff
@@ -31,33 +31,46 @@ const adminUser = {
 };
 
 Cypress.Commands.add("authenticate", (userType, userCredentials) => {
-  let credentials = {};
+  cy.session([userType, userCredentials], () => {
+    cy.visit("/");
+    let credentials = {};
 
-  if (userType && userCredentials) {
-    /* eslint-disable-next-line no-console */
-    console.warn(
-      "If userType and userCredentials are both provided, userType is ignored and provided userCredentials are used."
-    );
-  } else if (userCredentials) {
-    credentials = userCredentials;
-  } else if (userType) {
-    switch (userType) {
-      case "adminUser":
-        credentials = adminUser;
-        break;
-      case "stateUser":
-        credentials = stateUser;
-        break;
-      default:
-        throw new Error("Provided userType not recognized.");
+    if (userType && userCredentials) {
+      /* eslint-disable-next-line no-console */
+      console.warn(
+        "If userType and userCredentials are both provided, userType is ignored and provided userCredentials are used."
+      );
+    } else if (userCredentials) {
+      credentials = userCredentials;
+    } else if (userType) {
+      switch (userType) {
+        case "adminUser":
+          credentials = adminUser;
+          break;
+        case "stateUser":
+          credentials = stateUser;
+          break;
+        default:
+          throw new Error("Provided userType not recognized.");
+      }
+    } else {
+      throw new Error("Must specify either userType or userCredentials.");
     }
-  } else {
-    throw new Error("Must specify either userType or userCredentials.");
-  }
 
-  cy.xpath(cognitoEmailInputField).type(credentials.email);
-  cy.xpath(cognitoPasswordInputField).type(credentials.password, {
-    log: false,
+    cy.xpath(cognitoEmailInputField).type(credentials.email);
+    cy.xpath(cognitoPasswordInputField).type(credentials.password, {
+      log: false,
+    });
+    cy.get(cognitoLoginButton).click();
+
+    /**
+     * Waits for cognito session tokens to be set in local storage before saving session
+     * This ensures reused sessions maintain these tokens
+     * We expect at least three for the id, access, and refresh tokens
+     * We don't explicitly check for token names because they change
+     */
+    cy.waitUntil(() =>
+      cy.window().then((window) => window.localStorage.length > 3)
```

Comment:
Does the prefix of the name change? I like the update using waitUntil, but I think the filter could be improved. Or am I being overzealous in the approach?

Reply:
The prefix should be the same. Can you think of a succinct way to check for it? I didn’t want to get into a forEach in here